src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vilkårsvurdering/AutomatiskSatteVilkårUtils.kt

```diff
@@ -0,0 +1,63 @@
+import no.nav.familie.ks.sak.common.util.DATO_LOVENDRING_2024
+import no.nav.familie.ks.sak.common.util.erSammeEllerEtter
+import no.nav.familie.ks.sak.kjerne.behandling.domene.Behandling
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.PersonResultat
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.Resultat
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.Vilkår
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.VilkårRegelsett
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.VilkårResultat
+import java.time.LocalDate
+
+fun lagAutomatiskGenererteVilkårForBarnetsAlder(
+    personResultat: PersonResultat,
+    behandling: Behandling,
+    fødselsdato: LocalDate,
+): List<VilkårResultat> {
+    val periodeFomBarnetsAlderLov2024 = fødselsdato.plusMonths(13)
+    val periodeTomBarnetsAlderLov2024 = fødselsdato.plusMonths(19)
+
+    val periodeFomBarnetsAlderLov2021 = fødselsdato.plusYears(1)
+    val periodeTomBarnetsAlderLov2021 = fødselsdato.plusYears(2)
+
+    val erTruffetAvRegelverk2021 = periodeFomBarnetsAlderLov2021.isBefore(DATO_LOVENDRING_2024)
+    val erTruffetAvRegelverk2024 = periodeTomBarnetsAlderLov2024.erSammeEllerEtter(DATO_LOVENDRING_2024)
+
+    val vilkårResultatEtterRegelverk2021 =
```

Comment:
For å slippe null-verdier kan du vurdere å skrive noe slikt:

```suggestion
val vilkårResultater : MutableList<VilkårResultat> = mutableListOf()
if (erTruffetAvRegelverk2021) {
    vilkårResultater.add(
        VilkårResultat(
            ...
        )
    )
}
if (erTruffetAvRegelverk2024) {
    vilkårResultater.add(
        VilkårResultat(
            ...
        )
    )
}
return vilkårResultater.toList()
```

Reply:
Men da må vi ha muterbare lister :/

Reply:
Synes også det blir mindre eksplisitt hva vi returnerer på slutten av funksjonen. Nå sier vi at vi returnerer vilkårResultaterEtterRegelverk2021 og/eller vilkårResultatEtterRegelverk2024 så lenge de eksisterer. 
Null blir håndtert av typesystemet til Kotlin og det garanterer at vi ikke har null i returverdien, så nullverdier burde ikke være et problem